components/molecules/ContributorProfileHeader/contributor-profile-header.tsx

```diff
@@ -43,15 +66,15 @@ const ContributorProfileHeader = ({ avatarUrl, githubName, isConnected }: Contri
           />
         </div>
         {isConnected && (
-          <div className="flex md:translate-y-0 translate-y-28 gap-3 flex-col md:flex-row items-center">
-            <a target="_blank" rel="noopener noreferrer" href={`https://github.com/${githubName}`}>
-              <MarkGithubIcon size={24} className="visible xs:invisible mr-2 mb-3" />
-              <Button className="invisible xs:visible px-5 py-2 mb-6 bg-white" variant="text">
-                View on GitHub
+          <div className="flex flex-col items-center gap-3 md:translate-y-0 translate-y-28 md:flex-row">
+            {username !== user?.user_metadata.user_name && (
+              <Button onClick={handleClick} className="px-10 py-2 mb-10 bg-white md:mb-6 " variant="text">
+                {isFollowing ? "Unfollow" : "Follow"}
```

Comment:
This button should not be enabled for non-signed in users.
![image](https://user-images.githubusercontent.com/52013393/228983479-756653c3-d8b8-4b49-b9f7-fb6fe6d45aaf.png)

I shouldn't be able to click this without signing in. Would be even better if possible to force a sign in.

Reply:
Nice catch! 👍